blackjax/vi/svgd.py

```diff
@@ -0,0 +1,123 @@
+import functools
+from typing import Any, Callable, Dict, NamedTuple, Tuple
+
+import jax
+import jax.numpy as jnp
+from jax import grad, vmap
+from jax.flatten_util import ravel_pytree
+from jax.tree_util import tree_map
+
+from blackjax.base import Optimizer
+from blackjax.types import Array, PyTree
+
+
+def particles_to_array(particles_pytree: PyTree) -> Tuple[Array, Callable]:
+    particle_array = vmap(lambda p: ravel_pytree(p)[0])(particles_pytree)
+    single_particle = tree_map(lambda p: p[0], particles_pytree)
+    _, unravel_fn = ravel_pytree(single_particle)
+    return particle_array, unravel_fn
+
+
+def array_to_particles(particle_array: Array, unravel_fn: Callable) -> PyTree:
+    return jax.vmap(lambda p: unravel_fn(p))(particle_array)
+
+
+class SVGDState(NamedTuple):
+    particles: PyTree
+    kernel_parameters: Dict[str, PyTree]
+    opt_state: Any
+
+
+def init(
+    initial_particles: PyTree, kernel_parameters: Dict[str, Any], optimizer: Optimizer
+) -> SVGDState:
+    """
+    Initializes Stein Variational Gradient Descent Algorithm.
+
+    Parameters
+    ----------
+    initial_particles
+        Initial set of particles to start the optimization
+    kernel_paremeters
+        Arguments to the kernel function
+    optimizer
+        Optax compatible optimizer, which conforms to the `Optimizer` protocol
+    """
+    particle_array, _ = particles_to_array(initial_particles)
```

Comment:
you might need to extract the number of particles once here and include it in `SVGDState`